doc/source/development/debugging_extensions.rst

```diff
@@ -0,0 +1,51 @@
+.. _debugging_c_extensions:
+
+{{ header }}
+
+**********************
+Debugging C extensions
+**********************
+
+Pandas uses select C extensions for high performance IO operations. In case you need to debug segfaults or general issues with those extensions, the following steps may be helpful. These steps are geared towards using lldb as a debugger, though the steps for gdb will be similar.
+
+First, be sure to compile the extensions with the appropriate flags to generate debug symbols and remove optimizations. This can be achieved as follows:
+
+.. code-block:: sh
+
+   python setup.py build_ext --inplace -j4 --with-debugging-symbols
+
+Next you can create a script that hits the extension module you are looking to debug and place it in the project root. Thereafter launch a Python process under lldb:
+
+.. code-block:: sh
+
+   lldb python
+
+If desired, set breakpoints at various file locations using the below syntax:
+
+.. code-block:: sh
+
+   breakpoint set --file pandas/_libs/src/ujson/python/objToJSON.c --line 1547
+
+At this point you may get *WARNING:  Unable to resolve breakpoint to any actual locations.*. If you have not yet executed anything it is possible that this module has not been loaded into memory, which is why the location cannot be resolved. You can simply ignore for now as it will bind when we actually execute code.
+
+Finally go ahead and execute your script:
+
+.. code-block:: sh
+
+   run <the_script>.py
+
+Code execution will halt at the breakpoint defined or at the occurance of any segfault. LLDB's `GDB to LLDB command map <https://lldb.llvm.org/use/map.html>`_ provides a listing of debugger command that you can execute using either debugger.
+
+Another option to execute the entire test suite under the debugger would be to run the following:
+
+.. code-block:: sh
+
+   lldb -- python -m pytest
+
+Or for gdb
+
+.. code-block:: sh
+
+   gdb --args python -m pytest
```

Comment:
```
$ gdb --args python3 -m pytest
[...]
"~/.pyenv/shims/python3": not in executable format: File format not recognized
(gdb) run
Starting program:  -m pytest
No executable file specified.
Use the "file" or "exec-file" command.
```

Reply:
Can you try `gdb -ex r --args python3 -m pytest`? Taking that from this link:

https://wiki.python.org/moin/DebuggingWithGdb

Reply:
Yea looks like `gdb -ex r --args python3 -m pytest pandas/tests` was working for me if you want to try on yours and confirm

Reply:
```
$ gdb -ex r --args python3 -m pytest pandas/tests
[...]
"~/.pyenv/shims/python3": not in executable format: File format not recognized
Starting program:  -m pytest pandas/tests
No executable file specified.
Use the "file" or "exec-file" command.
(gdb) 
```

Reply:
Are you using pyenv for development or Conda?

Reply:
pyenv

Reply:
you've already gone above and beyond helping me debug this; ill spend some more time on this and ping you if i find anything new

Reply:
All good - I think this is generally helpful to hash out together so thanks for the input.

It looks like this might be specific to pyenv and how it manages the python executable:

https://stackoverflow.com/questions/48141135/cannot-start-dbg-on-my-python-c-extension

Reply:
`gdb -ex r --args bash pytest pandas/tests --skip-slow --skip-db` tentatively looks like a winner

Reply:
is the command from my previous comment specific to my case, or relevant to the document?

Reply:
I'm trying to avoid adding too much detail here since this issue is more of a pyenv thing than a debugger issue